src/main/java/org/broadinstitute/hellbender/utils/dataflow/SmallBamWriter.java

```diff
@@ -75,4 +98,71 @@ public void processElement(ProcessContext c) throws Exception {
             }
         }
     }
+
+    private static void writeToHadoop(Pipeline pipeline, PCollection<GATKRead> reads, final SAMFileHeader header, final String destPath) {
+        if (destPath.equals("/dev/null")) {
+            return;
+        }
+
+        String headerString = Base64.getEncoder().encodeToString(SerializableUtils.serializeToByteArray(header));
+
+        @SuppressWarnings("unchecked")
+        Class<? extends FileOutputFormat<NullWritable, SAMRecordWritable>> outputFormatClass =
+                (Class<? extends FileOutputFormat<NullWritable, SAMRecordWritable>>) (Class<?>) TemplatedKeyIgnoringBAMOutputFormat.class;
+        @SuppressWarnings("unchecked")
+        HadoopIO.Write.Bound<NullWritable, SAMRecordWritable> write = HadoopIO.Write.to(destPath,
+                outputFormatClass, NullWritable.class, SAMRecordWritable.class)
+                .withConfigurationProperty(TemplatedKeyIgnoringBAMOutputFormat.SAM_HEADER_PROPERTY_NAME, headerString);
+
+        reads.apply(ParDo.of(new DoFn<GATKRead, KV<NullWritable, SAMRecordWritable>>() {
+            private static final long serialVersionUID = 1L;
+            @Override
+            public void processElement(ProcessContext c) throws Exception {
+                SAMRecord samRecord = c.element().convertToSAMRecord(header);
+                SAMRecordWritable samRecordWritable = new SAMRecordWritable();
+                samRecordWritable.set(samRecord);
+                c.output(KV.of(NullWritable.get(), samRecordWritable));
+            }
+        })).setCoder(KvCoder.of(WritableCoder.of(NullWritable.class), WritableCoder.of(SAMRecordWritable.class)))
```

Comment:
The chaining makes this a little confusing on what's actually be operated on at each transform. Please add a comment explaining the different steps.